### PR TITLE
doc: blorb about `npm_config_devdir`

### PR DESCRIPTION
**Pull request by refack**
Fixes: #1180

### CODE REVIEW

**Comment by gibfahn** [comment]
I'm pretty sure this works for any of the `--option_name`, you just set `npm_config_option_name`, so we can just state it generically.

Also maybe worth mentioning that if you do `npm config set option_name val` then it only works if you run `node-gyp` through `npm`, but that if you do `export npm_config_option_name=val` or `set npm_config_option_name=val` then it works everywhere.

**Comment by refack** [comment]
@gibfahn PTAL

File: README.md [hunk 1]
```diff
@@ -192,6 +192,17 @@ Command Options
...
 
+Configuration
+--------
+
+#### `node-gyp` responds to configuration settings:
+Settings take the form `npm_config_OPTION_NAME` for any of the options listed above.<br>
+They can be set in two ways:
+1. via `npm config` - these work only when `node-gyp` is triggered by `npm`<br>
+   e.g. `$ npm config set npm_config_devdir /tmp/.gyp`
+2. as environment variables - work also when `noge-gyp` is invoked diretly<br>
```
Comment (richardlau) [inline README.md]: Typos, `noge-gyp`->`node-gyp` and `diretly`->`directly`.

File: README.md [hunk 2]
```diff
@@ -192,6 +192,17 @@ Command Options
...
+   e.g. `c:\> set npm_config_devdir=c:\temp\.gyp`
```
Comment (gibfahn) [inline README.md]: Maybe add an 

>`$ export npm_config_devdir=/tmp/.gyp` or 
>`c:\> set npm_config_devdir=c:\temp\.gyp` on Windows

File: README.md [hunk 3]
```diff
@@ -192,6 +192,17 @@ Command Options
 | `--solution=$solution`            | Set Visual Studio Solution version (win)
 
 
+Configuration
+--------
+
+#### `node-gyp` responds to configuration settings:
```
Comment (bnoordhuis) [inline README.md]: s/configuration settings/environment variables/?  Also please insert a blank line.

Comment (refack) [inline README.md]: I'm not sure either, but see if it works.

**Comment by refack** [comment]
@gibfahn @bnoordhuis fixed. PTAL

File: README.md [hunk 4]
```diff
@@ -192,6 +192,22 @@ Command Options
 | `--solution=$solution`            | Set Visual Studio Solution version (win)
 
 
+Configuration
+--------
+
+#### `node-gyp` responds to environment variables:
```
Comment (gibfahn) [inline README.md]: Still missing Ben's requested newline after this

File: README.md [hunk 5]
```diff
@@ -192,6 +192,22 @@ Command Options
...
+Variables take the form `npm_config_OPTION_NAME` for any of the options
+listed above.<br>
```
Comment (gibfahn) [inline README.md]: Why the `<br>`? If you want a line break just add two trailing spaces (or leave a blank line between for a proper paragraph break). In any case I think the next line can be part of this one.

File: README.md [hunk 6]
```diff
@@ -192,6 +192,22 @@ Command Options
...
+They can be set in two ways:
+1. as `npm` configuration - these work only when `node-gyp` is triggered by
+ `npm`. for example:<br>
```
Comment (gibfahn) [inline README.md]: ~`<br>`~

File: README.md [hunk 7]
```diff
@@ -192,6 +192,22 @@ Command Options
...
+ `$ npm config set [--global] npm_config_devdir /tmp/.gyp`
+2. as environment variables - these work also when `node-gyp` is invoked
+ directly:<br>
+ `> set npm_config_devdir=c:\temp\.gyp` for Windows or<br>
```
Comment (gibfahn) [inline README.md]: ~`<br>`~

File: README.md [hunk 8]
```diff
@@ -192,6 +192,22 @@ Command Options
...
+
+#### `node-gyp` responds to environment variables:
+Variables take the form `npm_config_OPTION_NAME` for any of the options
+listed above.<br>
+They can be set in two ways:
+1. as `npm` configuration - these work only when `node-gyp` is triggered by
+ `npm`. for example:<br>
+ `$ npm config set [--global] npm_config_devdir /tmp/.gyp`
+2. as environment variables - these work also when `node-gyp` is invoked
+ directly:<br>
```
Comment (gibfahn) [inline README.md]: ~`<br>`~

File: README.md [hunk 9]
```diff
@@ -192,6 +192,22 @@ Command Options
...
+ `> set npm_config_devdir=c:\temp\.gyp` for Windows or<br>
+ `$ export npm_config_devdir=/tmp/.gyp` for unix like environments
```
Comment (bnoordhuis) [inline README.md]: A few remarks:

- is the `####` at the top necessary?
- are the `<br>` tags necessary?
- maybe s/triggered/executed/?  'triggered' is somewhat unspecific.
- s/for example/For example/
- s/unix like/UNIX-like/
- insert a line or two explaining dashes should be replaced with underscores

Comment (gibfahn) [inline README.md]: I'd rather just have this as the default and then have the Windows one `for Windows`, that way people don't have to know whether they're in a Unix like environment, they just use default unless they're on Windows.

> `$ export npm_config_devdir=/tmp/.gyp` or `> set npm_config_devdir=c:\temp\.gyp` for Windows

Comment (refack) [inline README.md]: Wanted to mix it up for windows Falks #629, we tend to suffer most from `node-gyp` issues

Comment (refack) [inline README.md]: > is the #### at the top necessary?
 In the other suctions the sub-title is followed by a table. I thought here it should be delineated in some way 

> maybe s/triggered/executed/? 'triggered' is somewhat unspecific.

`npm` -> `node-gyp` is sort of a trigger... but I changed it anyway.

File: README.md [hunk 10]
```diff
@@ -192,6 +192,22 @@ Command Options
 | `--solution=$solution`            | Set Visual Studio Solution version (win)
 
 
+Configuration
+--------
+
+__`node-gyp` responds to environment variables:__
+Variables take the form `npm_config_OPTION_NAME` for any of the options
+listed above (dashes in option names should be replaced by an underscores).
```
Comment (richardlau) [inline README.md]: ~~an~~

File: README.md (hunk 9 above)
Comment (gibfahn) [inline README.md]: I'm not saying that Windows is less important, just that I think it's clearer to have the Unix method as the default. 

>`$ export npm_config_devdir=/tmp/.gyp` or  
>`> set npm_config_devdir=c:\temp\.gyp` for Windows

If you're on a mac, you might not know that you're on a Unix-like environment. But you do know you're not on Windows, so you end up with the right command.

Comment (refack) [inline README.md]: Changed.
(me 😭😭😭)

File: README.md [hunk 11]
```diff
@@ -192,6 +192,23 @@ Command Options
...
+Configuration
+--------
+
+__`node-gyp` responds to environment variables:__
+Variables take the form `npm_config_OPTION_NAME` for any of the options
+listed above (dashes in option names should be replaced by underscores).
+They can be set in two ways:
+1. as `npm` configuration - these work only when `node-gyp` is executed by
+ `npm`. For example:  
+ `$ npm config set [--global] npm_config_devdir /tmp/.gyp`
```
Comment (addaleax) [inline README.md]: Shouldn’t this be `npm config set [--global] devdir /tmp/.gyp`?

Comment (refack) [inline README.md]: Great catch!
Shame on us...🤦

**Comment by refack** [comment]
@richardlau @gibfahn Had rephrase for @addaleax's correction. PTAL.

**Comment by gibfahn** [comment]
@bnoordhuis LGTY?

**Comment by refack** [comment]
Just FMI was that a "Squash Merge". It looks really good on `master`.

**Comment by addaleax** [comment]
@refack there’s a lot of trailing whitespace here, btw… for next time :)

**Comment by refack** [comment]
> @refack there’s a lot of trailing whitespace here, btw… for next time :)

Ack. I got used to my IDE trimming them, but it doesn't for markdown because they might be semantic.
That said all the trainings are intentional for formating, except for the one in [line 199](https://github.com/nodejs/node-gyp/commit/c307b302f749f59c24fdd483acb32480571194b2#diff-04c6e90faac2675aa89e2176d2eec7d8R199)

**Pull request merged**